### DISCUSSION TITLE
Example AQL queries

### DISCUSSION THREAD

**Discussion by abhineet13** [AbsaOSS/spline#1025] [Q&A] [answered]
Hi, are there any reference AQL queries to interact with Collections of Spline DB.

**Comment by wajda** [comment, marked as answer]
There is no specific documentation as the Spline database isn't supposed to be accessed directly by 3rd party services. Although technically you can, but it is strongly discouraged. If you do so, you'd need to make sure your queries adhere to Spline internal conventions, particularly the ones related to ensuring a proper isolation level of application managed transactions (that will be introduced in Spline 1.0 due to lack of ACID capability in ArangoDB cluster deployments). Also upgrading to newer Spline releases would be more problematic as you'd have to update your queries manually if the db schema changes.

**Reply by wajda** [reply]
Here's the doc for the current (Spline 0.7) database schema - https://github.com/AbsaOSS/spline/blob/gh-pages/docs/Persistent%20model.pdf
We also use Foxx services for some more complex queries (e.g. end-2-end lineage query) - https://github.com/AbsaOSS/spline/tree/release/0.7.5/persistence/src/main/resources/foxx/spline
